addons/mail/static/src/components/chatter/chatter_suggested_recipient_tests.js

```diff
@@ -318,6 +321,59 @@ QUnit.test("more than 3 suggested recipients -> click 'show more' -> 'show less'
     );
 });
 
+/**
+ * Check that if partners created from suggestions do not match the suggestion, they are removed
+ */
+QUnit.test("new partners not matching suggested recipient are removed", async function (assert) {
+    assert.timeout(5000)
+    assert.expect(5);
+
+    const NEW_RES_ID = 20;
+
+    const openDialog = testUtils.makeTestPromise();
+    testUtils.mock.patch(ComposerSuggestedRecipient.prototype, {
+        _onDialogSaved: function(record) {
+            this._super(record);
+            openDialog.resolve(true);
+        },
+        _onChangeCheckbox: function() {
+            this._onDialogSaved({res_id: NEW_RES_ID});
+        }
+    });
+    this.data['res.fake'].records.push({
+        id: 10,
+        email_cc: "john@test.be",
+    });
+    const unlinked = testUtils.makeTestPromise();
+    await this.start({
+        mockRPC: function (route, args) {
+            if (route === '/web/dataset/call_kw/res.partner/unlink') {
+                assert.strictEqual(args.method, 'unlink', 'Should be deleting');
+                assert.strictEqual(args.model, 'res.partner', 'Should delete a partner');
+                assert.strictEqual(args.args[0], NEW_RES_ID, 'Should delete the new partner id');
+                unlinked.resolve(true);
+            }
+            else if (route === '/mail/get_suggested_recipients') {
+                return Promise.resolve({"10": [[false, "john@test.be", "Customer Email"]]});
+            }
+            return this._super(...arguments);
+        },
+    });
+    const chatter = this.env.models['mail.chatter'].create({
+        threadId: 10,
+        threadModel: 'res.fake',
+    });
+    await this.createChatterComponent({ chatter });
+    await afterNextRender(() => {
+        document.querySelector(`.o_ChatterTopbar_buttonSendMessage`).click();
+    });
+    document.querySelector('.o_ComposerSuggestedRecipient input[type=checkbox]').click();
+
+    const timeoutPromise = async (promise) => await Promise.any([promise, setTimeout(() => Promise.resolve(false), 5000)]);
+    assert.ok(await timeoutPromise(openDialog), 'Ticking the checkbox should have opened a "new parner" form dialog');
+    assert.ok(await timeoutPromise(unlinked), 'New partner not found by suggestion update, should deleted the record');
+});
+
```

Comment:
Do you have any suggestions to make the test less reliant on mocking? It's not like it's an urgent fix, I'd rather have a test than no test if possible. 

Thanks for the quick review.

Reply:
In `16.0` / `master`, can open a view with chatter and just `click()` where needed.
Might still need some mocking between the dialog is hardly testable without mock.

`/mail/get_suggested_recipients` might need some IMP in `MockServer` so it follows closely the behaviour of actual server rather than mocking in test and pretending the actual data being returned. It's still a **Mock**Server, but makes test more maintainable because `MockServer` existence is to have very close logic to actual server that is available in test environment.

Will see how actually testable the code is. If it's too mocked, then I much prefer no test at all. Reasoning is that mocking is cheating: the code is not testable, but you pretend it's testable by changing `X`, `Y`, and assumes server would provide `Z` when making this RPC (note that the actual RPC and its name is leaked, so test now relies on some implementation details).

I'm pretty sure the dialog for partner is not testable, so all forced attempts to have a test will be too mocked.
Hence why I think it's best have no test. We'll do effort in master to make this code testable, depending on time and priority, but that shouldn't prevent actually fixing the issue 😉